bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SortedLedgerStorage.java

```diff
@@ -144,9 +144,7 @@ public boolean ledgerExists(long ledgerId) throws IOException {
         // the O(1) for the ledgerCache.
         if (!interleavedLedgerStorage.ledgerExists(ledgerId)) {
             EntryKeyValue kv = memTable.getLastEntry(ledgerId);
-            if (null == kv) {
-                return interleavedLedgerStorage.ledgerExists(ledgerId);
```

Comment:
I have seen this patch before.
Maybe there is already another open PR

My question is...
Why this code was written the way it is?
In order to deal with some possible race condition ?

@merlimat do you know?

Reply:
Is there any case that line 145 returns `false`, but line 148 returns `true`?